source/Plugins.cpp

```diff
@@ -0,0 +1,124 @@
+/* Plugins.cpp
+Copyright (c) 2022 by Sam Gleske (samrocketman on GitHub)
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#include "Plugins.h"
+
+#include "DataFile.h"
+#include "DataNode.h"
+#include "DataWriter.h"
+#include "Files.h"
+
+#include <algorithm>
+#include <map>
+
+using namespace std;
+
+namespace {
+	map<string, bool> settings;
+	map<string, bool> frozenSettings;
+
+	void LoadSettings(const string &path)
+	{
+		DataFile prefs(path);
+		for(const DataNode &node : prefs)
+		{
+			const string &key = node.Token(0);
+			if(key == "plugins")
+				for(const DataNode &child : node)
+					if(child.Size() == 2)
+						settings[child.Token(0)] = child.Value(1);
```

Comment:
If you load the settings directly into `frozenSettings` you don't need the `Freeze` function.

Reply:
I think logic would be simpler with the two maps being equivalent.  Otherwise, I'll need to use `map.find` and search for `map.end()` when doing a one to one comparison between two maps.

Is that what you mean?  If not, can you elaborate?

Reply:
Adding on, even if `plugins.txt` exist, if a user added a new plugin the entry would not exist in persisted settings the first launch.  So Freeze is still required in that case as well.

Reply:
Removed frozenSettings in the latest commit in favor of a bool pair.  The freeze is still necessary but it no longer does a map copy.